tests/jballerina-unit-test/src/test/resources/test-src/types/jsontype/json-value.bal

```diff
@@ -432,3 +432,59 @@ function testMapJsonToJsonArray() {
     json[] j = [2];
     map<json>[] p = <map<json>[]> j;
 }
+
+function testJsonLaxErrorLifting() returns error? {
+    check testSimpleVarRefJsonErrorLiftingInLaxTyping();
+    check testJsonMapErrorLifingInLaxTyping();
+    check testJsonReturnTypeErrorLifting();
+}
```

Comment:
Shall we break this into three different tests and call them separately from the Java side? 

Also since this may return error, shouldn't we have asserted something on the Java side anyway? Instead we could use checkpanic here, to make the test fail if the function returns error.

Reply:
Shouldn't we assert in Ballerina code itself?

Reply:
We should, but since `check` just returns the error and we are not asserting anything on the Java side, this test wouldn't fail if any of these functions returned error, right?

But if we do something like
```ballerina
function testJsonLaxAccess1() {
    checkpanic testSimpleVarRefJsonErrorLiftingInLaxTyping();
}

function testJsonLaxAccess2() {
    checkpanic testJsonMapErrorLifingInLaxTyping();
}

function testJsonLaxAccess3() {
    checkpanic testJsonReturnTypeErrorLifting();
}
```

for the positive cases (i.e., where they shouldn't return error), and call these `testJsonLaxAccessN` functions from Java, if the function incorrectly returns error the Java test will fail.

Or we should do
```ballerina
function testJsonLaxAccess1() {
    error? res = testSimpleVarRefJsonErrorLiftingInLaxTyping();
    assertTrue(res is ());
}
```

For the runtime negative cases (which I believe we haven't added yet), we have to do this anyway, and assert that `res is error`.